### ISSUE TITLE
No isError property on curried tasks?

### ISSUE THREAD

**Issue by mwpastore** [machty/ember-concurrency#258]
I have a page that shows a bunch of address cards, and a task that takes an address model as its input and attempts to destroy it. I am currying the task for each address model and attempting to show when each one is working or is errored. The .isRunning property works as expected, but the .isError property does not. 

```hbs
{{#each addresses as |address|}}
  <Address @address=address />

  {{#with (task deleteTask address) as |task|}}
    <button onclick={{perform task}}>Delete</button>

    {{#if task.isRunning}}
      Working... {{! this works great! }}
    {{else if task.isError}}
      Oops! {{! this does not }}
    {{//if}}
  {{/with}}
{{/each}}
```

When I inspect the curried task in the console, I see .isIdle, .isQueued, and .isRunning properties, but no .isError property. I can't use `task.last.error` or `task.lastErrored` because those point to the un-curried task instance. In other words, if one curried deleteTask fails, the "Oops!" message shows for every address. 

I will refactor this to use a component for each address, and hang the deleteTask off the component, but it would be nice if this worked as expected.

**Comment by maxfierke** [comment]
Hi @mwpastore 👋 ! Apologies for the late reply on this. The reason for this is that `isError` exists only on a `TaskInstance` and not a `Task`. A `Task` itself is never in an error state, but individual instances of one might be.

I think the `task.last` "pointing to the un-curried task", is the real root of the issue.

When a task is curried, it is cloned, but the task's scheduler is not cloned, which is where state around `last*` lives. _That_ does strike me as confusing behavior, because it would seem expected that currying would perhaps unlink these and provide a new "realm" for task state.

**tl;dr:** it's either a bug or confusing semantics for curried tasks that should be documented. Perhaps future @machty can weigh in, but I'm going to mark this a bug for now.
